consistence-coding-standard.md

```diff
@@ -337,33 +352,64 @@ class X
 
 new X(
 	$foo,
-	$bar
+	$string
 );
 ```
 
-* Default argument values are used only when needed - to express optional argument (only at the end of the list) or to allow passing null to a type hinted argument.
+* Default argument values are used only when needed to either express optional argument (only at the end of the list) or to allow passing null to a type hinted argument.
+  * If possible (PHP 7.1+), nullable types (`?string`) are used to allow passing null to a type hinted argument and therefore default arguments are used only for optional arguments.
+  * For scalar arguments default arguments are used only for optional arguments, not to allow passing nulls (see detailed example below).
 
 ```php
 <?php
 
+// for PHP 7.0
+
 class X
 {
 
 	/*
-	 * @param \Foo $a
-	 * @param \Foo|null $b
-	 * @param string $c
-	 * @param string|null $d
-	 * @param string $e
-	 * @param string|null $f
+	 * @param \Foo $a required type argument
+	 * @param \Foo|null $b required argument, but nullable type needed
+	 * @param string $c required scalar argument
+	 * @param string|null $d required argument with nullable scalar
+	 * @param string $e optional nullable scalar argument
+	 * @param string|null $f optional nullable scalar argument
 	 */
 	public function __construct(
 		Foo $a,
 		Foo $b = null,
-		$c,
+		string $c,
 		$d,
-		$e = '',
-		$f = null
+		string $e = '',
+		string $f = null
+	)
+	{
+		// ...
+	}
+
+}
+
+// or with PHP 7.1+
+
+class X
+{
+
+	/*
```

Comment:
Myslíš potřeba z hlediska vysvětlení, nebo CS?

Z hlediska vysvětlení je tam imho potřeba, aby byly vidět ty typy vs nullable types a defaulty u obojího (s tím, že PHPDoc se oproti 7.0 nemění, ale co je kde potřeba psát, to jo).

A z hlediska CS (jakože "nepiš zbytečné PHPDocs"), tak je taky imho potřeba, protože tam jsou právě descriptions těch jednotlivých parametrů, co který reprezentuje.